server.js

```diff
@@ -12,16 +12,121 @@ mongoose.Promise = Promise;
 const port = process.env.PORT || 8080;
 const app = express();
 
+// Create a list of all endpoints
+const listEndpoints = require('express-list-endpoints');
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
+
+// Schemas and models start here
+const { Schema } = mongoose;
+
+const ThoughtSchema = new Schema({
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  heart: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: () => Date.now()
+  }
+}); 
+
+const Thought = mongoose.model("Thought", ThoughtSchema);
+
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.status(200).send({
+    succes: true,
+    message: "OK",
+    body: {
+      content: "Johannas Happy Thoughs API",
+      endpoints: listEndpoints(app)
+    }
+  });
+});
+
+// Finding thoughts, sort them in descening (desc) order by creation date and limit thoughts to 20 and execute query
+app.get("/thoughts", async (req, res) => {
+  const thoughts = await Thought.find()
+  .sort({createdAt: 'desc', _id: 'desc' })
+  .limit(20)
+  .exec();
+  res.status(200).json(thoughts);
+});
+
+app.get("/thoughts/id/:id", async (req, res) => {
+  const { id } = req.params;
+  try {
+    const singleThought = await Thought.findById(id)
+    if (singleThought) {
+      res.status(200).json(singleThought);
+    } else {
+      res.status(400).json('Not found');
+    }
+    } catch (e) {
+      res.status(400).json('No id found');
+    }
+});
+
+
+app.post("/thoughts", async (req, res) =>{
+  const { message } = req.body;
+    try {
+      const thought = await new Thought({ message }).save();
+      res.status(201).json({
+        succes: true,
+        response: thought,
+        message: "Created with success"
+      });
+    } catch (e) {
+      res.status(400).json({
+        succes: false,
+        response: e,
+        message: "Error occurd whilst posting your thought"
+      });
+    }
 });
```

Comment:
Nice success / error messages, even though I think 400 in general stands for "internal service error"